conda/common/logic.py

```diff
@@ -263,6 +263,20 @@ def process_solution(self, sat_solution):
         return solution
 
 
+_sat_solver_str_to_cls = {
+    "pycosat": _PycoSatSolver,
+    "pycryptosat": _PyCryptoSatSolver,
+    "pysat": _PySatSolver,
+}
+
+_sat_solver_cls_to_str = {cls: string for string, cls in _sat_solver_str_to_cls.items()}
+
+
+PycoSatSolver = "pycosat"
```

Comment:
The plain `str` here may seem a little strange. This is just to decouple the `common.logic` -> `common._logic` dependency a bit more, meaning the former only needs to import `Clauses, FALSE, TRUE` from the latter and nothing more. That is helpful when we drop-in replace `common._logic` by some other implementation.

Reply:
We'll have to be very careful to make sure that people don't import anything else.  Is it worthwhile to mark everything else in that file as _, or is that just dumb overkill?

Reply:
I think it's probably adequate to just have it over in _logic.py.

Reply:
I'm not understanding completely. Can you clarify?
Do you mean marking the other members (apart from `Clauses, FALSE, TRUE`) of `_logic` as private via `_`?

Reply:
Addressed in https://github.com/conda/conda/pull/9427/commits/2aae4ed0822770a9d82e950026e6bf3076b17415.